#!/usr/bin/env python
PACKAGE = "vpa_vision2action"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("vf", double_t, 0, "linear speed x when going straight", 0.35, 0, 1)
gen.add("vs", double_t, 0, "steering linear speed", 0.35, 0, 0.5)
gen.add("kp_lf", double_t, 0, "p gain for lane following maintain", 5, 0, 20)
gen.add("ki_lf", double_t, 0, "i gain for lane following maintain", 0, 0, 20)

gen.add("kp_acc", double_t, 0, "p gain for ACC", 0.05, 0, 20)
gen.add("ki_acc", double_t, 0, "i gain for ACC", 0, 0, 20)
gen.add("acc_ref", int_t, 0, "desired distance of acc", 60, 0, 200)

exit(gen.generate(PACKAGE, "vpa_vision2action", "lane_follow_pi"))
